### PR TITLE
Introduce and use Matrix<>::swap

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/libs/antares/array/matrix.hxx [hunk 1]
```diff
@@ -1552,6 +1552,18 @@ inline void Matrix<T, ReadWriteT>::copyFrom(const Matrix<U, V>* rhs)
         copyFrom(*rhs);
 }
 
+template<class T, class ReadWriteT>
+template<class U, class V>
+void Matrix<T, ReadWriteT>::swap(Matrix<U, V>& rhs)
+{
+  static_assert(Yuni::Static::Type::StrictlyEqual<T, U>::Yes);
+  using std::swap;
```
Comment (guilpier-code) [inline src/libs/antares/array/matrix.hxx:1560]: You don't include **algorithm.h** for std::swap, meaning that it's included elsewhere... Strange

File: src/libs/antares/array/matrix.hxx [hunk 2]
```diff
@@ -1552,6 +1552,18 @@ inline void Matrix<T, ReadWriteT>::copyFrom(const Matrix<U, V>* rhs)
         copyFrom(*rhs);
 }
 
+template<class T, class ReadWriteT>
+template<class U, class V>
+void Matrix<T, ReadWriteT>::swap(Matrix<U, V>& rhs)
```
Comment (guilpier-code) [inline src/libs/antares/array/matrix.hxx]: The type of **rhs** is different from local matrix's type. How can it be ?

File: src/libs/antares/array/matrix.hxx [hunk 3]
```diff
@@ -1552,6 +1552,18 @@ inline void Matrix<T, ReadWriteT>::copyFrom(const Matrix<U, V>* rhs)
...
+{
+  static_assert(Yuni::Static::Type::StrictlyEqual<T, U>::Yes);
+  using std::swap;
+  swap(this->width, rhs.width);
```
Comment (guilpier-code) [inline src/libs/antares/array/matrix.hxx:1561]: If we swap matrices, their width and height must be equal, right ?
If not, we have a problem, don't you think ?

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/libs/antares/array/matrix.hxx (hunk 1 above)
Comment (flomnes) [reply src/libs/antares/array/matrix.hxx:1560]: I have no idea how `std::swap` is defined here. However I included header `utility` for safety.

File: src/libs/antares/array/matrix.hxx (hunk 3 above)
Comment (flomnes) [reply src/libs/antares/array/matrix.hxx:1561]: `swap` should always succeed. Adding an assertion could potentially make it fail.

swapping matrices of different size makes sense. It is up to the user to check for size consistency.

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**